src/Compiler/Driver/fsc.fs

```diff
@@ -850,7 +873,7 @@ let main3
             | Some ib -> ib.RawMetadata.TryGetILModuleDef().Value.MetadataVersion
             | _ -> ""
 
-    let optimizedImpls, optDataResources =
+    let optimizedImpls, (optDataResources, optHash) =
```

Comment:
It's probably better to separate the hash generation from the encodeOptimizationData method.

Reply:
Will check, did not do it initially because of doubled-processing that would be needed.

Reply:
Done